### PR TITLE
Hopefully fix duplicate issues properly.

### PR DESCRIPTION
**Pull request by SM91337**
Remove previous duplicate fix as it breaks chat message being shown to sender.
Rewrite Protobuf messages instead of creating a new one.
Create one usermessage to all recipients instead of multiple for BitBuffer.
Check recipients array to see if the recipient already exists before we add him.
Make sure sender is always in the recipients array.

### CODE REVIEW

File: scripting/chat-processor.sp [hunk 1]
```diff
@@ -279,14 +294,16 @@ public void Frame_OnChatMessage_SayText2(any data)
...
+		if(!bProto) {
+			Handle hMsg = StartMessage("SayText2", iRecipients, iNumRecipients, USERMSG_RELIABLE|USERMSG_BLOCKHOOKS);
+			
+			BfWriteByte(hMsg, iSender);
+			BfWriteByte(hMsg, bChat);
+			BfWriteString(hMsg, sBuffer);
+			EndMessage();
+		
+		} else {
+			PbSetString(bfMsg, "msg_name", sBuffer);
```
Comment (peace-maker) [inline scripting/chat-processor.sp]: Did you try copying the value of "textallchat" from the original protobuf message when creating your own here?

Might have to do some observations on the value if some consumer changes sFlag in the forward, so we adjust the value correctly.

Comment (SM91337) [inline scripting/chat-processor.sp]: This should not be needed as we are modifying the message only, so the other keys should be left untouched as they originally were.

File: scripting/chat-processor.sp [hunk 2]
```diff
@@ -152,23 +151,25 @@ public Action OnSayText2(UserMsg msg_id, BfRead msg, const int[] players, int pl
...
 	for (int i = 0; i < playersNum; i++)
 	{
-		PushArrayCell(hRecipients, players[i]);
+		if(FindValueInArray(hRecipients, players[i]) == -1) 
+		{
+			PushArrayCell(hRecipients, players[i]);
+		}
+	}
+	
+	if(FindValueInArray(hRecipients, iSender) == -1)
```
Comment (SM91337) [inline scripting/chat-processor.sp]: I am not entirely sure what you meant here, I took a wild guess in my latest pull :P

File: scripting/chat-processor.sp.rej [hunk 3]
```diff
@@ -0,0 +1,15 @@
+diff a/scripting/chat-processor.sp b/scripting/chat-processor.sp	(rejected hunks)
```
Comment (SM91337) [inline scripting/chat-processor.sp.rej]: SourceTree added it for some reason, I removed it in next pull.

**Pull request merged**